ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -2180,6 +2180,82 @@ private void setUpKsqlResource() {
     ksqlResource.configure(ksqlConfig);
   }
 
+  @Test
+  public void shouldThrowOnDenyListedStreamProperty() {
+    // Given:
+    ksqlResource = new KsqlResource(
+        ksqlEngine,
+        commandStore,
+        DISTRIBUTED_COMMAND_RESPONSE_TIMEOUT,
+        activenessRegistrar,
+        (ec, sc) -> InjectorChain.of(
+            schemaInjectorFactory.apply(sc),
+            topicInjectorFactory.apply(ec),
+            new TopicDeleteInjector(ec, sc)),
+        Optional.of(authorizationValidator),
+        errorsHandler
+    );
+    final Map<String, Object> props = new HashMap<>(ksqlRestConfig.getKsqlConfigProperties());
+    props.put(KsqlConfig.KSQL_PROPERTIES_OVERRIDES_DENYLIST,
+        StreamsConfig.NUM_STREAM_THREADS_CONFIG);
+    ksqlResource.configure(new KsqlConfig(props));
+
+    // When:
+    final EndpointResponse response = ksqlResource.handleKsqlStatements(
+        securityContext,
+        new KsqlRequest(
+            "query",
+            ImmutableMap.of(StreamsConfig.NUM_STREAM_THREADS_CONFIG, "1"), // stream properties
+            emptyMap(), // config properties
+            null
+        )
+    );
+
+    // Then:
+    assertThat(response.getStatus(), CoreMatchers.is(BAD_REQUEST.code()));
+    assertThat(((KsqlErrorMessage) response.getEntity()).getMessage(),
+        is("A property override was set locally for a property that the server prohibits "
+            + "overrides for: '" + StreamsConfig.NUM_STREAM_THREADS_CONFIG + "'"));
+  }
+
+  @Test
+  public void shouldThrowOnDenyListedConfigProperty() {
+    // Given:
+    ksqlResource = new KsqlResource(
+        ksqlEngine,
+        commandStore,
+        DISTRIBUTED_COMMAND_RESPONSE_TIMEOUT,
+        activenessRegistrar,
+        (ec, sc) -> InjectorChain.of(
+            schemaInjectorFactory.apply(sc),
+            topicInjectorFactory.apply(ec),
+            new TopicDeleteInjector(ec, sc)),
```

Comment:
Would be much easier to test if you inject it. Then all you need to do is verify its being called with the right params, i.e. one simple test.

Reply:
Done. I injected the DenyListPropertyValidator.